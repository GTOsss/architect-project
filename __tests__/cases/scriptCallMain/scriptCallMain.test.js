const appRoot = process.cwd();
const { resolve } = require('path');
const {
  returnOutputPathsWithContent,
  promisifyCliCommand,
  SnapshotSerializer,
} = require('../../utils/utilsForArcStart');
const { pathForCommand } = require('../casesConfigPath');

const currentCommand = pathForCommand.scriptCallMain;

describe('scriptCallMain', () => {
  test('test call main function from script', async () => {
    await promisifyCliCommand(`arc s -c ${currentCommand}`);
    const currentPathToOutput = resolve(appRoot, currentCommand, 'output');

    SnapshotSerializer();

    expect(returnOutputPathsWithContent(currentPathToOutput)).toMatchSnapshot();
  });
});
