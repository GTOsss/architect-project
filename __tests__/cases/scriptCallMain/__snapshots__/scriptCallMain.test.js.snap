// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scriptCallMain test call main function from script 1`] = `
Object {
  "src": Array [
    Object {
      "content": "test",
      "path": "*/scriptCallMain/output/src/services/icons/svg/a.svg",
    },
    Object {
      "content": "test",
      "path": "*/scriptCallMain/output/src/services/icons/svg/arrow-right.svg",
    },
    Object {
      "content": "test",
      "path": "*/scriptCallMain/output/src/services/icons/svg/blue-posts.svg",
    },
    Object {
      "content": " const options = {
    escapeString: false,
    indent: 2,
    escapeRegex: false,
  };

  const t = "t"
",
      "path": "*/scriptCallMain/output/src/services/name/index.ts",
    },
  ],
}
`;
